lib/kennel/models/record.rb

```diff
@@ -148,6 +148,52 @@ def invalid_update!(field, old_value, new_value)
 
       private
 
+      def validate_json_respecting_filters(data)
+        caught = begin
+                   validate_json(data)
+                   nil
+                 rescue ValidationError => e
+                   e
+                 end
+
+        if caught.nil?
+          unless skip_validations.empty?
+            invalid! nil, "skip_validations must not be used when no validations need skipping"
+          end
+
+          unless validate
+            if ENV["PROJECT"] || ENV["TRACKING_ID"]
+              invalid! nil, "validate must not be set to false when no validations need skipping"
+            elsif ENV["SHOW_DISABLED_VALIDATION"]
+              Kennel.out.puts "#{tracking_id} I: validate must not be set to false when no validations need skipping"
+            end
+          end
+        else
+          unless validate
+            if caught.tag
+              if ENV["PROJECT"] || ENV["TRACKING_ID"]
+                invalid! nil, "`validate: false` is deprecated. Replace `validate: false` by `skip_validations: [#{caught.tag.inspect}]`"
+              elsif ENV["SHOW_DISABLED_VALIDATION"]
+                Kennel.out.puts <<~MESSAGE
+                  #{tracking_id} W: `validate: false` is deprecated. Replace `validate: false` by `skip_validations: [#{caught.tag.inspect}]`
+                MESSAGE
+              end
+            end
+
+            return
+          end
+
+          if skip_validations.include?(caught.tag)
+            # Ideally we'd check that skip_validations contains _only_ those
+            # validations while fail. However because the validation errors
+            # are raised exceptions, we can only see up to 1 per record, so
+            # we can't do that.
```

Comment:
we could mark "used skips" and then compare against that
... but ideally we don't raise exceptions and then discard them since that's pretty slow, but make the validation calls do nothing if they are skipped (and mark "skip used" right there too)